soc/arm/nuvoton_npcx/npcx7/soc.h

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2020 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _NUVOTON_NPCX_SOC_H_
+#define _NUVOTON_NPCX_SOC_H_
+
+/* CMSIS required definitions */
+#define __FPU_PRESENT  CONFIG_CPU_HAS_FPU
```

Comment:
no objection to this, but, normally, these should be defined in vendor HAL header (which includes the cm4.h cmsis header)

Reply:
Hi @ioannisg:
Since there is no HAL repository for nuvoton npcx ec, I referred to these examples in Zephyr repository (https://github.com/zephyrproject-rtos/zephyr/blob/master/soc/arm/ti_simplelink/cc13x2_cc26x2/soc.h#L28 and https://github.com/zephyrproject-rtos/zephyr/blob/master/soc/arm/arm/mps2/soc.h#L14) and put the definitions used for CMSIS in soc.h.